cli/command/image/build.go

```diff
@@ -344,9 +360,18 @@ func runBuild(dockerCli *command.DockerCli, options buildOptions) error {
 	// Everything worked so if -q was provided the output from the daemon
 	// should be just the image ID and we'll print that to stdout.
 	if options.quiet {
-		fmt.Fprintf(dockerCli.Out(), "%s", buildBuff)
+		imageID = fmt.Sprintf("%s", buildBuff)
+		fmt.Fprintf(dockerCli.Out(), "%s", imageID)
 	}
 
+	if options.imageIDFile != "" {
+		if imageID == "" {
+			return errors.Errorf("Server did not provide an image ID. Cannot write %s", options.imageIDFile)
+		}
+		if err := ioutil.WriteFile(options.imageIDFile, []byte(imageID), 0666); err != nil {
+			return err
+		}
+	}
```

Comment:
The function is fairly long, but it seems to me to be fairly straightforward and sequential. I don't think moving the code for handling this one option out of line will aid clarity (in fact I think it would obfuscate things for no real benefit) and I don't think the code to instantiate the new struct, call its methods when relevant and handle any resulting errors will substantially reduce the amount of code added to this function by this PR.